src/pages/posts/[slug].js

```diff
@@ -147,26 +147,28 @@ export async function getStaticProps({ params = {} } = {}) {
   const socialImage = `${process.env.OG_IMAGE_DIRECTORY}/${params?.slug}.png`;
 
   const { categories, databaseId: postId } = post;
-  const category = categories.length && categories[0];
-  let { name, slug } = category;
+
+  const { category: relatedCategory, posts: relatedPosts } = await getRelatedPosts(categories, postId);
 
   return {
     props: {
       post,
       socialImage,
       relatedPosts: {
-        posts: await getRelatedPosts(category, postId),
+        posts: relatedPosts,
         title: {
-          name: name || null,
-          link: categoryPathBySlug(slug),
+          name: relatedCategory.name || null,
+          link: categoryPathBySlug(relatedCategory.slug),
```

Comment:
I just found a problem here for **uncategorized** posts: it will fail trying to parse the static props as the `relatedPosts` can be an undefined value. Also it would need an optional chaining to access name and slug from `relatedCategory`. 

One possible solution:
```
relatedPosts: {
        posts: relatedPosts || [],
        title: {
          name: relatedCategory?.name || null,
          link: relatedCategory?.slug ? categoryPathBySlug(relatedCategory.slug) : null,
        },
      },
```
I also realized that the post property needs and empty array fallback just to ensure the `!!relatedPostsList.length`  validation on **_line 115_** (or an optional chaining there, whatever you think makes more sense)

Reply:
great catch. ended up going with: ```const { category: relatedCategory, posts: relatedPosts } = await getRelatedPosts(categories, postId);
  const hasRelated = relatedCategory && Array.isArray(relatedPosts) && relatedPosts.length;
  const related = !hasRelated
    ? null
    : {
        posts: relatedPosts,
        title: {
          name: relatedCategory.name || null,
          link: categoryPathBySlug(relatedCategory.slug),
        },
      };

  return {
    props: {
      post,
      socialImage,
      related,
    },
  };```

Reply:
Nice!